### ISSUE TITLE
`pixi shell` should only pull the requested environment, not all environments.

### ISSUE THREAD

**Issue by sebastian-sz** [prefix-dev/pixi#2114]
### Problem description

TL;DR `pixi shell` pulls ALL environments, even when only the default is requested.

I'm rolling a Pytorch CPU / GPU project. `linux-64` uses GPU and CPU, everything else is CPU. Please find the `pyproject.toml` below:
```toml
[project]
authors = [{name = "Sebastian Sz", email = "***@***.com"}]
dependencies = ["jaxtyping>=0.2.34,<0.3"]
description = "Add a short description here"
name = "torch-proj"
requires-python = ">= 3.9, <3.13"
version = "0.1.0"

[build-system]
build-backend = "hatchling.build"
requires = ["hatchling"]

[tool.pixi.project]
channels = ["conda-forge", "pytorch"]
platforms = ["linux-64", "osx-arm64", "win-64"]

[tool.pixi.dependencies]
pytorch = {version = "2.4.1", channel = "pytorch"}
torchvision = {version = "0.19.1", channel = "pytorch" }

[tool.pixi.pypi-dependencies]
torch-proj = { path = ".", editable = true }

[tool.pixi.tasks]

[tool.pixi.environments]
cuda = ["cuda"]

# CUDA setup
[tool.pixi.feature.cuda]
platforms = ["linux-64"]
channels = ["nvidia/label/cuda-12.4.0", "nvidia", "conda-forge", "pytorch"]

[tool.pixi.feature.cuda.dependencies]
pytorch-cuda = {version = "12.4.*", channel = "pytorch"}
```

The issue is: when running `pixi shell` on Linux, the `cuda` environment gets downloaded and installed, even though I did not request cuda. This is kinda bloaty, e.g. when I want to run CI on Linux on the CPU there is no reason to pull the Nvidia's packages (which are quite heavy).

Ideally, we would pull cuda only when requested, e.g:  
`pixi shell` <- downloads CPU only packages
`pixi shell -e cuda` <- downloads CUDA only packages

Rather then have everything possible downloaded, regardless of the environment.

Pixi version: `0.30.0`

**Comment by ruben-arts** [comment]
The problem your having is caused by our requirement to install the conda environments to solve for the pypi dependencies. What you require for this to work is the feature requested in: https://github.com/prefix-dev/pixi/issues/1046

What you could try for now is a `pip install --editable .` task, instead of the following configuration:
```
[tool.pixi.pypi-dependencies]
torch-proj = { path = ".", editable = true }
```
This would avoid the installation to only when you need it instead of requiring the environment at solve time.

**Comment by sebastian-sz** [comment]
@ruben-arts Thank you for the response.

It indeed only pulls a single env when I comment out the `[tool.pixi.pypi-dependencies]` and also the `dependencies` in `[project]` section inside the `pyproject.toml`.

I am closing the issue as there is no need for duplicate, to track this.
